### DISCUSSION TITLE
Fullpage Cache is enabled by default

### DISCUSSION THREAD

**Discussion by thgross** [pimcore/pimcore#10610] [Q&A / Support] [answered]
Beginning with pimcore 10, the fullpage cache is enabled by default, but only in the `prod`-environment. However, since this was not the case in earlier versions of pimcore, this can lead to quite some problems.
Reading the [Full Page Cache Documentation](https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Cache/Full_Page_Cache.html) leads to the assumption that you need to configure the cache to enable it.
Also, [the Upgrade Guide](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10.html#page_Adapt-Security-config) mentions the cache setting, but doesn't tell us about the change from disabled to enabled.

pimcore has some logic to automatically disable caching, for example:
- in `dev`-environments
- when you are logged into the pimcore backend
- when a session is active
- when it is not a GET request

The problem really hits when your application uses an `Authorization`-header instead of sessions to identify it's users. Everything works fine in the dev environment and since you assume that the full page cache is disabled anyway, as soon as you deploy to your prod-environment, every user get's the same data from the api:
`GET ​/api​/user​/me`

**So my main question is**: Is it a bug or a feature that the full page cache is enabled by default? If it is not a bug, it should absolutely be mentioned in the documentation.

**Comment by brusch** [comment, marked as answer]
@thgross actually it's intentional that the full page cache is enabled in production, but you're right, we should consider  `Authorization`  headers. Can you create a PR?

**Reply by thgross** [reply]
Thank you for your answer.

I'm ready to create a PR soon.
